domain/services/model.go

```diff
@@ -314,39 +308,45 @@ func (s *ModelFactory) UnitState() *unitstateservice.Service {
 	)
 }
 
-// CloudImageMetadata returns the service for persisting and retrieving cloud image metadata for the current model.
+// CloudImageMetadata returns the service for persisting and retrieving cloud
+// image metadata for the current model.
 func (s *ModelFactory) CloudImageMetadata() *cloudimagemetadataservice.Service {
 	return cloudimagemetadataservice.NewService(
-		cloudimagemetadatastate.NewState(changestream.NewTxnRunnerFactory(s.controllerDB), s.clock, s.logger.Child("cloudimagemetadata")),
+		cloudimagemetadatastate.NewState(
+			changestream.NewTxnRunnerFactory(s.controllerDB),
+			s.clock,
+			s.logger.Child("cloudimagemetadata"),
+		),
 	)
 }
 
 // Port returns the service for managing opened port ranges for units.
 func (s *ModelFactory) Port() *portservice.WatchableService {
-	logger := s.logger.Child("port")
 	return portservice.NewWatchableService(
 		portstate.NewState(changestream.NewTxnRunnerFactory(s.modelDB)),
-		domain.NewWatcherFactory(s.modelDB, logger.Child("watcherfactory")),
-		logger.Child("service"),
+		s.modelWatcherFactory("port"),
+		s.logger.Child("port"),
+	)
+}
+
+// BlockCommand returns the service for blocking commands.
+func (s *ModelFactory) BlockCommand() *blockcommandservice.Service {
+	return blockcommandservice.NewService(
+		blockcommandstate.NewState(changestream.NewTxnRunnerFactory(s.modelDB)),
+		s.logger.Child("blockcommand"),
 	)
 }
 
 // Stub returns the stub service. A special service which collects temporary
 // methods required to wire together domains which are not completely implemented
 // or wired up.
 //
+// *** ADD NEW METHODS ABOVE THIS, NOT BELOW.
```

Comment:
This should be a comment below the Stub method.

Reply:
Then you can add above the comment and below the stub :)